File: gosnack_admin/src/lib/firebase/firestore/collections.ts
/** Constantes de nomes de coleções raízes do Firestore. */
export const FirestoreRootCollections = {
  /** Coleção de categorias. */
  CATEGORIES: "categories",

  /** Coleção de restrições alimentares. */
  DIETARY_RESTRICTIONS: "dietaryRestrictions",

  /** Coleção de ingredientes. */
  INGREDIENTS: "ingredients",

  /** Coleção de pedidos realizados. */
  ORDERS: "orders",

  /** Coleção de unidades escolares. */
  UNITS: "units",

  /** Coleção de informações dos usuários. */
  USERS: "users",
} as const

/** Constantes de nomes de subcoleções no Firestore. */
export const FirestoreSubCollections = {
  // Subcoleções de "units/{unitId}/..."

  /** Coleção de produtos e pratos do cardápio. */
  PRODUCTS: "products",

  /** Coleção de lanchonetes. */
  SNACK_BARS: "snackBars",
} as const

File: gosnack_client/lib/utils/constants/firestore/collections.dart
/// Nomes das coleções raiz do Firestore.
abstract class FirestoreCollections {
  // -- Root Collections ---------------------------------------------------- //

  /// Coleção de categorias.
  static const String categories = 'categories';

  /// Coleção de ingredientes.
  static const String ingredients = 'ingredients';

  /// Coleção de pedidos.
  static const String orders = 'orders';

  /// Coleção de unidades escolares.
  static const String units = 'units';

  /// Coleção de dados dos usuários.
  static const String users = 'users';

  // -- Subcollections ------------------------------------------------------ //

  /// Subcoleção de lanchonetes dentro de uma unidade.
  static const String cafeterias = 'cafeterias';

  /// Subcoleção de produtos dentro de uma lanchonete.
  static const String products = 'products';
}
